drivers/frequency/adf4368/adf4368.h

```diff
@@ -0,0 +1,661 @@
+
+#define ADF4368_SPI_SCRATCHPAD_TEST		0x5A
+
+/* Specifications */
+#define ADF4368_SPI_WRITE_CMD			0x0
+#define ADF4368_SPI_READ_CMD			0x8000
+#define ADF4368_SPI_DUMMY_DATA			0x00
+#define ADF4368_BUFF_SIZE_BYTES			3
+#define ADF4368_VCO_FREQ_MIN			6400000000U	// 6.4GHz
+#define ADF4368_VCO_FREQ_MAX			12800000000U	// 12.8GHz
+#define ADF4368_MOD1WORD			0x2000000U	// 2^25
+#define ADF4368_MOD2WORD_MAX			0xFFFFFFU	// 2^24 - 1
+#define ADF4368_PHASE_RESYNC_MOD2WORD_MAX	0x1FFFFU	// 2^17 - 1
+#define ADF4368_CHANNEL_SPACING_MAX		78125U
+#define ADF4368_PFD_FREQ_MAX			625000000U	// 625MHz
+#define ADF4368_PFD_FREQ_FRAC_MAX		250000000U	// 250MHz
+#define ADF4368_DCLK_DIV1_0_MAX			160000000U	// 160MHz
+#define ADF4368_DCLK_DIV1_1_MAX			320000000U	// 320MHz
+#define ADF4368_CLKOUT_DIV_REG_VAL_MAX		3
+
+#define ADF4368_RFOUT_MAX			12800000000U // 12.8GHz
+#define ADF4368_RFOUT_MIN			800000000U // 0.8GHz
+#define ADF4368_REF_CLK_MAX			4000000000U // 4GHz
+#define ADF4368_REF_CLK_MIN			10000000 // 10MHz
+#define ADF4368_REF_DIV_MAX			63
+#define ADF4368_OUT_PWR_MAX			15
+#define ADF4368_CPI_VAL_MAX			15
+#define ADF4368_BLEED_WORD_MAX			8191
+#define ADF4368_FRAC_N_INT_MIN			19
+#define ADF4368_INT_N_INT_MIN			4
+
+#define ADF4368_PHASE_BLEED_CNST		2044000
+#define ADF4368_BLEED_N_INT_TH			35
+#define ADF4368_COARSE_BLEED_CNST		202 // 202ua
+#define ADF4368_FINE_BLEED_CNST1		250 // 250ua
+#define ADF4368_FINE_BLEED_CNST2		512 // 250ua
+#define ADF4368_FINE_BLEED_CNST3		567 // 567na
+
+#define ADF4368_SIGMA_DELTA_MOD_CNST		4096 // 250ua
+
+#define ADF4368_POR_DELAY_US			200
+#define ADF4368_LKD_DELAY_MS			9
+
+#define MHZ					MEGA
+#define S_TO_NS					NANO
+#define NS_TO_PS				KHZ_PER_MHZ
+#define PS_TO_NS				KHZ_PER_MHZ
+#define PS_TO_US				MICRO
+#define US_TO_MS				MILLI
+#define MS_TO_S					MILLI
+#define PS_TO_S					PICO
+#define US_TO_FS				NANO
+#define NS_TO_S					NANO
+
+/**
+ * @struct adf4368_init_param
+ * @brief ADF4368 Initialization Parameters structure.
+ */
+struct adf4368_init_param {
+	/** SPI Initialization parameters */
+	struct no_os_spi_init_param	*spi_init;
+	bool				spi_4wire_en;
+	bool				cmos_3v3;
+	bool				ref_doubler_en;
+	uint8_t				ref_div;
+	uint8_t				cp_i;
+	uint8_t				ld_count;
+	uint64_t			ref_freq_hz;
+	uint64_t			freq;
+};
+
+/**
+ * @struct adf4368_dev
+ * @brief ADF4368 Device Descriptor.
+ */
+struct adf4368_dev {
+	/** SPI Descriptor */
+	struct no_os_spi_desc		*spi_desc;
+	bool				spi_4wire_en;
+	bool				cmos_3v3;
+	bool				phase_pol;
+	bool				ref_doubler_en;
+	uint8_t				ref_div;
+	uint8_t				cp_i;
+	uint8_t				ld_count;
+	uint8_t				clkout_div_reg_val_max;
+	uint16_t 			n_int;
+	int16_t				bleed_word;
+	uint32_t			phase_adj;
+	uint64_t			vco_max;
+	uint64_t			vco_min;
+	uint64_t			freq_max;
+	uint64_t			freq_min;
+	uint64_t			ref_freq_hz;
+	uint64_t			freq;
+};
+
+/**
+ * @struct reg_sequence
+ * @brief ADF4368 register format structure for default values
+ */
+struct reg_sequence {
```

Comment:
You can make this static. Better yet to move it and the default values to the driver source file. Also, aren't these register values automatically loaded after a reset.

Reply:
This values are the recommended register values for given initial state. After a reset chip's default register values are different.

Struct definition is not static but struct array itself is static in the header file.

I wanted to make the driver similar to adf4382 driver but if its better in the source file I can . 
You mean static struct reg_sequence in the header file and  move the struct array to source file, right?

Reply:
structure and array relocated to source file.